### PR TITLE
refactor: add / use 'Client._post_resource' method

### PR DESCRIPTION
**Pull request by tseaver**
Toward #38.

~~Note: this PR is based on 6244df0 of PR #442.  I will rebase to master when that PR lands.~~ Done.

### CODE REVIEW

File: google/cloud/storage/client.py [hunk 1]
```diff
@@ -528,6 +528,77 @@ def _put_resource(
...
         )
 
+    def _post_resource(
+        self,
+        path,
+        data,
+        query_params=None,
+        headers=None,
+        timeout=_DEFAULT_TIMEOUT,
+        retry=DEFAULT_RETRY,
```
Comment (andrewsg) [inline google/cloud/storage/client.py:538]: In my concurrent retry work I've been following a policy of defaulting to None on internal methods, esp. those related to POST which frequently are not idempotent. User-facing will have sensible defaults (often conditional ones). This is just a convention I've adopted and it's certainly open to disagreement.

File: google/cloud/storage/notification.py [hunk 2]
```diff
@@ -233,7 +233,7 @@ def _set_properties(self, response):
         self._properties.clear()
         self._properties.update(response)
 
-    def create(self, client=None, timeout=_DEFAULT_TIMEOUT):
+    def create(self, client=None, timeout=_DEFAULT_TIMEOUT, retry=DEFAULT_RETRY):
```
Comment (andrewsg) [inline google/cloud/storage/notification.py:236]: Not idempotent, must have no retry argument or retry=None default.

File: google/cloud/storage/client.py [hunk 3]
```diff
@@ -1278,6 +1348,7 @@ def create_hmac_key(
         project_id=None,
         user_project=None,
         timeout=_DEFAULT_TIMEOUT,
+        retry=DEFAULT_RETRY,
```
Comment (andrewsg) [inline google/cloud/storage/client.py:1351]: Not idempotent, must have no retry argument or retry=None default.

Do you have access to our GCS Consolidated Retry Strategy document? If not, we need to get you in on that.

**Review by andrewsg — CHANGES_REQUESTED** [review]

File: google/cloud/storage/client.py (hunk 1 above)
Comment (tseaver) [reply google/cloud/storage/client.py]: 1b081c8f1e5a2f27eb4c7254adc09ead21356120

File: google/cloud/storage/notification.py (hunk 2 above)
Comment (tseaver) [reply google/cloud/storage/notification.py]: 7c1f93dde1765bfb2744660498485649ca2153b7

File: google/cloud/storage/client.py (hunk 3 above)
Comment (tseaver) [reply google/cloud/storage/client.py]: a7bd535715c3e22d921920170828dedda163d721

**Review by chrisrossi — APPROVED** [review]
LGTM!

**Review by andrewsg — APPROVED** [review]
Thanks!

**Pull request merged**